### PR TITLE
Add method to join neighboring prompt items in a Prompt

### PR DESCRIPTION
**Pull request by benbrandt**

### CODE REVIEW

File: src/prompt.rs [hunk 1]
```diff
@@ -29,6 +30,32 @@ impl<'a> Prompt<'a> {
     pub fn borrow(&'a self) -> Prompt<'a> {
         Self(self.0.iter().map(|item| item.borrow()).collect())
     }
+
+    /// When constructing prompts programatically, it can be beneficial to append several
+    /// text items in a prompt. For example, if doing a fewshot prompt as the first item,
+    /// and user input as a second item.
```
Comment (pacman82) [inline src/prompt.rs:36]: Great comment

File: src/prompt.rs [hunk 2]
```diff
@@ -29,6 +30,32 @@ impl<'a> Prompt<'a> {
...
+    ///
+    /// Howver, because of how tokenization works, having each item tokenized separately
```
Comment (pacman82) [inline src/prompt.rs:38]: However

**Review by pacman82 — APPROVED** [review]
Great PR. Just some nitpicks

File: src/prompt.rs (hunk 1 above)
Comment (benbrandt) [reply src/prompt.rs:36]: Danke

**Pull request merged**